sklearn/metrics/tests/test_pairwise.py

```diff
@@ -473,7 +473,7 @@ def check_pairwise_distances_chunked(X, Y, working_memory, metric='euclidean'):
                                      metric=metric)
     assert isinstance(gen, GeneratorType)
     blockwise_distances = list(gen)
-    Y = np.array(X if Y is None else Y)
+    Y = X if Y is None else Y
```

Comment:
I think this change is necessary otherwise the zeroing logic in ``pairwise_distances`` doesn't kick in.